### PR TITLE
fix(docs/rules): add missing category tags to rules metadata

### PR DESCRIPTION
**Pull request by mDalessandro**
Add category tags to the rules that were missing them.
Add validation to check during a build for the inclusion of a category on the rule metadata.

### Newly added categories for rules which were missing them:

**aria-input-field-name**: `cat.aria`
**aria-toggle-field-name**: `cat.aria`
**aria-tooltip-name**: `cat.aria`
**avoid-inline-spacing**: `cat.structure`
**identical-links-same-purpose**: `cat.semantics`
**label-content-name-mismatch**: `cat.semantics`
**no-autoplay-audio**: `cat.time-and-media`
**scrollable-region-focusable**: `cat.keyboard`

Let me know if you have a different thought on which categories should be used for these rules. Because there's no description of the categories themselves, it wasn't clear which one should be used; for example, both `cat.semantics` and `cat.structure` seemed to have some overlap. That could probably be a new gihub issue if anyone else feels similarly.

Closes issue #2554

## Reviewer checks

**Required fields, to be filled out by PR reviewer(s)**

- [x] Follows the commit message policy, appropriate for next version
- [x] Code is reviewed for security

### CODE REVIEW

File: lib/rules/accesskeys.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
 	"id": "accesskeys",
 	"selector": "[accesskey]",
 	"excludeHidden": false,
-	"tags": ["best-practice", "cat.keyboard"],
+	"tags": ["cat.keyboard", "best-practice"],
```
Comment (mDalessandro) [inline lib/rules/accesskeys.json:5]: I swapped the positioning here to better match the order of the other rules' tags.

File: build/tasks/validate.js [hunk 2]
```diff
@@ -198,6 +198,10 @@ function createSchemas() {
 				type: 'array',
 				items: {
 					type: 'string'
+				},
+				conform: function hasCategoryTag(tags) { return tags.some(tag => tag.includes('cat.')) },
```
Comment (straker) [inline build/tasks/validate.js:202]: Nice. I like this check. Thanks for adding this.

**Review by straker — COMMENTED** [review]
Look great! Just want @WilcoFiers to verify the categories are correct as he knows them best.

**Comment by straker** [comment]
@mDalessandro so we merged a new rule and now this pr has a conflict. Sorry about that. Would you mind merging the latest develop to resolve the conflict?

**Comment by WilcoFiers** [comment]
@straker no need. It's an easy fix. I've sorted it.

**Review by WilcoFiers — APPROVED** [review]

**Comment by mDalessandro** [comment]
Awesome, thanks for taking care of that. I haven't had a chance to sit down and do it, so that was helpful.

I'll keep an eye out for any other issues I could work on.

**Comment by padmavemulapati** [comment]
Added `category` - tag for the following rules/ids

aria-input-field-name: cat.aria
aria-toggle-field-name: cat.aria
aria-tooltip-name: cat.aria
avoid-inline-spacing: cat.structure
identical-links-same-purpose: cat.semantics
label-content-name-mismatch: cat.semantics
no-autoplay-audio: cat.time-and-media
scrollable-region-focusable: cat.keyboard
accesskeys: cat.keyboard

**Pull request merged**